src/gentropy/ukb_ppp_eur_sumstat_preprocess.py

```diff
@@ -0,0 +1,115 @@
+from gentropy.datasource.ukb_ppp_eur.summary_stats import UkbPppEurSummaryStats
+
+
+class UkbPppEurStep:
+    """UKB PPP (EUR) data ingestion and harmonisation."""
+
+    def __init__(
+        self, session: Session, raw_study_index_path: str, raw_summary_stats_path: str, variant_annotation_path: str, tmp_variant_annotation_path: str, output_path: str
+    ) -> None:
+        """Run UKB PPP (EUR) data ingestion and harmonisation step.
+
+        Args:
+            session (Session): Session object.
+            raw_study_index_path (str): Input raw study index path.
+            raw_summary_stats_path (str): Input raw summary stats path.
+            variant_annotation_path (str): Input variant annotation dataset path.
+            tmp_variant_annotation_path (str): Temporary output path for variant annotation dataset.
+            output_path (str): Output path.
+        """
+        session.logger.info("Pre-compute the direct and flipped variant annotation dataset.")
+        va_df = (
+            session
+            .spark
+            .read
+            .parquet(variant_annotation_path)
+        )
+        va_df_direct = (
+            va_df.
+            select(
+                f.col("chromosome").alias("vaChromosome"),
+                f.col("variantId"),
+                f.concat(
+                    f.col("chromosome"),
+                    f.lit("_"),
+                    f.col("position"),
+                    f.lit("_"),
+                    f.col("referenceAllele"),
+                    f.lit("_"),
+                    f.col("alternateAllele")
+                ).alias("ukb_ppp_id"),
+                f.lit("direct").alias("direction")
+            )
+        )
+        va_df_flip = (
+            va_df.
+            select(
+                f.col("chromosome").alias("vaChromosome"),
+                f.col("variantId"),
+                f.concat(
+                    f.col("chromosome"),
+                    f.lit("_"),
+                    f.col("position"),
+                    f.lit("_"),
+                    f.col("alternateAllele"),
+                    f.lit("_"),
+                    f.col("referenceAllele")
+                ).alias("ukb_ppp_id"),
+                f.lit("flip").alias("direction")
+            )
+        )
+        (
+            va_df_direct.union(va_df_flip)
+            .coalesce(1)
+            .repartition("vaChromosome")
+            .write
+            .partitionBy("vaChromosome")
+            .mode("overwrite")
+            .parquet(tmp_variant_annotation_path)
+        )
+
+        session.logger.info("Process study index.")
+        (
+            UkbPppEurStudyIndex.from_source(
+                spark=session.spark,
+                raw_study_index_path=raw_study_index_path,
+                raw_summary_stats_path=raw_summary_stats_path,
+            )
+            .df
+            .write
+            .mode("overwrite")
+            .parquet(f"{output_path}/study_index")
```

Comment:
Could we keep the name `study_index` directly in the in the output_path?

Reply:
Sorry, not sure I understand correctly... Because this step has two outputs, which should be under the same directory, I've defined the `output_path` as a basename, and then add `study_index` or `summary_stats` as required. Do you suppose I should just define two output paths for study index and summary stats instead?

Reply:
Fair enough, changes made: https://github.com/opentargets/gentropy/pull/652/commits/56dfd43f7620193348d72b0889b30cf723d9fd29